src/app/breed/breed.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+
+import { DogService } from '../dogs/dogs.service';
+
+@Component({
+	selector: 'app-breed',
+	templateUrl: './breed.component.html',
+	styleUrls: ['./breed.component.css']
+})
+export class BreedComponent implements OnInit {
+
+	public subBreeds: string[] = [];
+	public breedName: string = "";
+	public imagePath: string = "";
+
+	constructor(
+		private readonly dogService: DogService,
+		private readonly activatedRoute: ActivatedRoute,
+		private readonly router: Router
+	) {
+		this.activatedRoute.params.subscribe(
+			(params: Params) => {
+				this.breedName = params['name'];
+			}
+		);
+	}
+
+	public async ngOnInit() {
+		this.getSubBreeds(this.breedName);
+		this.getImageByBreed(this.breedName);
+	}
+
+	public getSubBreeds(breedName: string) {
```

Comment:
prepareSubBreedNames

get methods should return something according to best practices